src/hypertable.h

```diff
@@ -26,9 +26,24 @@ typedef struct SubspaceStore SubspaceStore;
 typedef struct Chunk Chunk;
 typedef struct Hypercube Hypercube;
 
-#define TS_HYPERTABLE_HAS_COMPRESSION(ht)                                                          \
-	((ht)->fd.compressed_hypertable_id != INVALID_HYPERTABLE_ID)
+/* For the distributed node case, we would have compression enabled
+ * but don't have a corresponding internal table on the access
+ * node
+ */
+enum
+{
+	HypertableCompressionOff = 0,
+	HypertableCompressionEnabled = 1,
+	HypertableInternalCompressionTable = 2,
+};
 
+#define TS_HYPERTABLE_HAS_COMPRESSION(ht) ts_hypertable_has_compression(ht)
+
+#define TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht)                                                  \
+	((ht)->fd.compression_state == HypertableCompressionEnabled)
```

Comment:
What is the difference between `TS_HYPERTABLE_HAS_COMPRESSION` and `TS_HYPERTABLE_HAS_COMPRESSION_ENABLED`? 

This is somewhat confusing and it is difficult to tell these two macros apart.

Reply:
TS_HYPERTABLE_HAS_COMPRESSION is used in places where we need to know that the internal table exists for the  user table. The action is done on the internal table.

Reply:
Sorry, I should have been more clear/specific. The issue here is that the macros are hard to tell apart. I understand that they are used differently, although it was initially hard for me to realize the difference. This leads me to believe that this will be the case for others too, so just explaining in the PR is not enough.

If I look at the implementation of `ts_hypertable_has_compression` it is asserted that `ht->fd.compression_state == HypertableCompressionEnabled` is true when the function returns true. This is the same condition used in `TS_HYPERTABLE_HAS_COMPRESSION_ENABLED`. So it is really unclear what the difference is.

I think what we actually want to do here is distinguish between  distributed and non-distributed compressed hypertables:

```
#define TS_HYPERTABLE_IS_DISTRIBUTED_AND_COMPRESSED(ht) \
   (TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht) && \
     ht->fd.compressed_hypertable_id != INVALID_HYPERTABLE_ID)

```

This would make the naming more clear and also, looking at the definition, that this is a stricter condition than just having compression enabled.

Reply:
The check is for the existence of a  "compressed" hypertable. I think a more appropriate rename is TS_HYPERTABLE_HAS_COMPRESSION_TABLE. If it is distributed, compressed_hypertable_id would be null.